File: Templates/TypeEditor/src/Templates/Templates/VB.TypeEditor/CustomControlLibrary.Server/CustomTypeEditor/CustomTypeEditorVM.Factory.vb
Imports CustomControlLibrary.Protocol
Imports Microsoft.DotNet.DesignTools.ViewModels

Partial Friend Class CustomTypeEditorVM
    ''' <summary>
    ''' Factory class which generates the CustomTypeEditorViewModel.
    ''' </summary>
    <ExportViewModelFactory(ViewModelNames.CustomTypeEditorVM)>
    Private Class Factory
        Inherits ViewModelFactory(Of CustomTypeEditorVM)

        Protected Overrides Function CreateViewModel(ByVal provider As IServiceProvider) As CustomTypeEditorVM
            Return New CustomTypeEditorVM(provider)
        End Function
    End Class
End Class

File: Templates/TypeEditor/src/TemplateSolutions/CS.CustomTypeEditor/CustomControlLibrary.Server/CustomTypeEditor/CustomTypeEditorVM.Factory.cs
using System;
using CustomControlLibrary.Protocol;
using Microsoft.DotNet.DesignTools.ViewModels;

namespace CustomControlLibrary.Designer.Server
{
    internal partial class CustomTypeEditorVM
    {
        /// <summary>
        /// Factory class which generates the CustomTypeEditorViewModel.
        /// </summary>
        [ExportViewModelFactory(ViewModelNames.CustomTypeEditorVM)]
        private class Factory : ViewModelFactory<CustomTypeEditorVM>
        {
            protected override CustomTypeEditorVM CreateViewModel(IServiceProvider provider)
                => new(provider);
        }
    }
}
